### PR TITLE
Update 'convert namespace' to only 'dedent' file contents, not format them.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/58078

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertNamespaceRefactoringTests.cs [hunk 1]
```diff
@@ -417,8 +417,9 @@ class C
 }
 ",
                 FixedCode = @"
-namespace $$N; // comment
+namespace $$N;
 
+// comment
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertNamespaceRefactoringTests.cs:422]: this result is more desirable than the prior one.

File: src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertNamespaceRefactoringTests.cs [hunk 2]
```diff
@@ -477,10 +478,9 @@ public async Task TextConvertToFileScopedWithCommentedOutContents()
 ",
                 FixedCode = @"
 namespace N;
-
-    // public class C
-    // {
-    // }
+// public class C
+// {
+// }
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertNamespaceRefactoringTests.cs:483]: Better result than before

File: src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertNamespaceRefactoringTests.cs [hunk 3]
```diff
@@ -513,7 +513,7 @@ public class C
 {
 }
 
-    // I'll probably write some more code here later
+// I'll probably write some more code here later
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertNamespaceRefactoringTests.cs:516]: Better result than before

File: src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertToFileScopedNamespaceAnalyzerTests.cs [hunk 4]
```diff
@@ -347,8 +347,9 @@ class C
 }
 ",
                 FixedCode = @"
-namespace $$N; // comment
+namespace $$N;
 
+// comment
 class C
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertToFileScopedNamespaceAnalyzerTests.cs:353]: Better result than before

File: src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertToFileScopedNamespaceAnalyzerTests.cs [hunk 5]
```diff
@@ -391,6 +392,290 @@ class C
...
+
+#if X
+class C
+{
+}
+#else
+class C
+{
+}
+#endif
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertToFileScopedNamespaceAnalyzerTests.cs:455]: We could argue that we should not touch the disabled section. there are pros/cons each way.

File: src/Workspaces/Core/Portable/Indentation/AbstractIndentationService.Indenter.cs [hunk 6]
```diff
@@ -171,7 +171,7 @@ public bool TryGetSmartTokenIndentation(out IndentationResult indentationResult)
 
                     var formatter = _service.CreateSmartTokenFormatter(this);
                     var changes = formatter.FormatTokenAsync(Document.Project.Solution.Workspace, token, CancellationToken)
-                                           .WaitAndGetResult(CancellationToken);
+                                           .WaitAndGetResult_CanCallOnBackground(CancellationToken);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Indentation/AbstractIndentationService.Indenter.cs:174]: we call this from teh BG.  so this is needed.  i looked into making this all async, but it was a PITA.

File: src/Features/Core/Portable/Wrapping/AbstractCodeActionComputer.cs [hunk 7]
```diff
@@ -104,13 +105,7 @@ protected string GetSmartIndentationAfter(SyntaxNodeOrToken nodeOrToken)
...
                     CancellationToken);
 
-                var baseLine = newSourceText.Lines.GetLineFromPosition(desiredIndentation.BasePosition);
-                var baseOffsetInLine = desiredIndentation.BasePosition - baseLine.Start;
-
-                var indent = baseOffsetInLine + desiredIndentation.Offset;
-
-                var indentString = indent.CreateIndentationString(UseTabs, TabSize);
-                return indentString;
+                return desiredIndentation.GetIndentationString(newSourceText, UseTabs, TabSize);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Wrapping/AbstractCodeActionComputer.cs:108]: extracted into an extension.

File: src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs [hunk 8]
```diff
@@ -26,21 +31,118 @@ namespace Microsoft.CodeAnalysis.CSharp.ConvertNamespace
...
+
+            // Next, actually replace the block namespace with the file scoped namespace.
+            var annotation = new SyntaxAnnotation();
+            var (updatedDocument, semicolonSpan) = await ReplaceWithFileScopedNamespaceAsync(document, namespaceDeclaration, annotation, cancellationToken).ConfigureAwait(false);
+
+            // Now, find the file scoped namespace in the updated doc and go and dedent every line if applicable.
+            if (indentation == null)
+                return (updatedDocument, semicolonSpan);
+
+            return await DedentNamespaceAsync(updatedDocument, indentation, annotation, cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs:65]: this method is the new logic for how things work.

File: src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs [hunk 9]
```diff
@@ -66,51 +168,47 @@ public static BaseNamespaceDeclarationSyntax Convert(BaseNamespaceDeclarationSyn
 
         private static FileScopedNamespaceDeclarationSyntax ConvertNamespaceDeclaration(NamespaceDeclarationSyntax namespaceDeclaration)
         {
-            // We move leading and trailing trivia on the open brace to just be trailing trivia on the semicolon, so we preserve
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs:69]: there was a lot of wonky logic here moving trivia around and adding elasticity.  THis has been changed to be much simpler, with no elastic trivia anymore.

File: src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs [hunk 10]
```diff
@@ -66,51 +168,47 @@ public static BaseNamespaceDeclarationSyntax Convert(BaseNamespaceDeclarationSyn
...
+            var semiColon = SyntaxFactory.Token(SyntaxKind.SemicolonToken).WithoutTrivia();
 
+            // Move trivia after the original name token to now be after the new semicolon token.
             var fileScopedNamespace = SyntaxFactory.FileScopedNamespaceDeclaration(
                 namespaceDeclaration.AttributeLists,
                 namespaceDeclaration.Modifiers,
                 namespaceDeclaration.NamespaceKeyword,
                 namespaceDeclaration.Name.WithoutTrailingTrivia(),
-                semiColon,
+                semiColon.WithTrailingTrivia(namespaceDeclaration.Name.GetTrailingTrivia()),
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs:179]: simpler logic than before. w hat was after the name should be after the semicolon.

File: src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs [hunk 11]
```diff
@@ -66,51 +168,47 @@ public static BaseNamespaceDeclarationSyntax Convert(BaseNamespaceDeclarationSyn
...
                 namespaceDeclaration.Externs,
                 namespaceDeclaration.Usings,
-                namespaceDeclaration.Members).WithAdditionalAnnotations(Formatter.Annotation);
+                namespaceDeclaration.Members);
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs:182]: no more formatting.  we control all dedentation ourselves.

File: src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs [hunk 12]
```diff
@@ -66,51 +168,47 @@ public static BaseNamespaceDeclarationSyntax Convert(BaseNamespaceDeclarationSyn
...
+            // If the open-brace token has any special trivia, then move them to before the first member in the namespace.
+            if (namespaceDeclaration.OpenBraceToken.LeadingTrivia.Any(t => t.IsSingleOrMultiLineComment() || t.IsDirective) ||
+                namespaceDeclaration.OpenBraceToken.TrailingTrivia.Any(t => t.IsSingleOrMultiLineComment() || t.IsDirective))
             {
                 fileScopedNamespace = fileScopedNamespace.ReplaceToken(
                     firstBodyToken,
-                    firstBodyToken.WithPrependedLeadingTrivia(SyntaxFactory.ElasticCarriageReturnLineFeed));
+                    firstBodyToken.WithPrependedLeadingTrivia(namespaceDeclaration.OpenBraceToken.GetAllTrivia()));
+                firstBodyToken = fileScopedNamespace.SemicolonToken.GetNextNonZeroWidthTokenOrEndOfFile();
+
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs:194]:
```suggestion
```

**Review by davidwengier — APPROVED** [review]
Nice one!

**Pull request merged**